### ISSUE TITLE
Improve pagination API for `client.getRooms()` endpoint in `@liveblocks/node`

### ISSUE THREAD

**Issue by nvie** [liveblocks/liveblocks#1382]
If you're using `@liveblocks/node` to iterate over all your rooms, i.e. `client.getRooms(...)`, there is some pagination support, i.e. `client.getRooms({ limit, startingAfter: cursor })`.

However, to fetch the next page, the only way currently to extract the cursor value to use on the next request is by parsing it out of the `nextPage` field manually, which is quite annoying and not a great DX:

```ts
let cursor: string | null = null;

while (true) {
  const resp = await client.getRooms({ startingAfter: cursor });

  doSomethingWithRooms(resp.data);  // current page

  if (resp.nextPage === null) {
    break; // Done paginating
  } else {
    // Ugh :(
    cursor = resp.nextPage.split("startingAfter=")[1].split("&")[0];
  }
}
```

It'd be nice to have a more convenient pagination API.

(Possibly the solution here is not to update the `@liveblocks/node` client, but to make the cursor available in a separate field in the server responses instead so at least we avoid the parsing.)

**Comment by saiprakash2** [comment]
Hii @nvie ,
I have one suggestion here. Can't we do this `cursor = resp.nextPage.split("startingAfter=")[1].split("&")[0];` in the api itself so that we can send cursor value in the server responses. In that way they dont' have to do the parsing.
If it is null then no more pagination or if it has value then they can use it to fetch the next set of data.
This might be an easy fix.
Please let me know if it works or I will look into it bit more 🙂.

**Comment by nvie** [comment]
Yep! That’s my preferred approach too. We can let the server send those `cursor` and `limit` fields back in those responses, right next to the `nextPage` field with that value in there.

**Comment by saiprakash2** [comment]
Hi @nvie, 
Are you talking about improving REST api server responses here instead of updating `liveblocks/node` client? 
Is REST api code open sourced?

**Comment by nvie** [comment]
> Are you talking about improving REST api server responses here instead of updating `liveblocks/node` client?

Exactly, it would be nicer to have the data available directly instead of having to parse it out in the node client, which feels like a bit of a hack!

> Is REST api code open sourced?

Unfortunately the REST API is not open source.

**Comment by CTNicholas** [comment]
Hey @saiprakash2, we've just released a new option that simplifies this. If we take the original code snippet, you can see the before and after in the snippet below, enabled by `nextCursor`.

```ts
let cursor: string | null = null;

while (true) {
  const resp = await client.getRooms({ startingAfter: cursor });
  doSomethingWithRooms(resp.data);  // current page

  if (resp.nextPage === null) {
    break; // Done paginating
  } else {
    // ❌ Before
    cursor = resp.nextPage.split("startingAfter=")[1].split("&")[0];

    // ✅ After
    cursor = resp.nextCursor;
  }
}
```

You can find more information under [`liveblocks.getRooms`](https://liveblocks.io/docs/api-reference/liveblocks-node#get-rooms).
